src/include/rbd/librbd.h

```diff
@@ -1476,6 +1490,20 @@ CEPH_RBD_API int rbd_group_snap_list(rados_ioctx_t group_p,
 CEPH_RBD_API int rbd_group_snap_list_cleanup(rbd_group_snap_info_t *snaps,
                                              size_t group_snap_info_size,
                                              size_t num_entries);
+CEPH_RBD_API int rbd_group_snap_list2(rados_ioctx_t group_p,
+                                      const char *group_name,
+                                      rbd_group_snap_info_v2_t *snaps,
+                                      size_t *num_entries);
+CEPH_RBD_API int rbd_group_snap_list2_cleanup(rbd_group_snap_info_v2_t *snaps,
+                                              size_t num_entries);
+CEPH_RBD_API  int  rbd_group_snap_info(
+    rados_ioctx_t group_p, const char *group_name, const char *snap_name,
+    rbd_group_snap_info_v2_t *group_snap,
+    rbd_group_image_snap_info_t *image_snaps, size_t *num_image_snaps);
```

Comment:
I'm not sure I'm following the rationale for having `rbd_group_snap_list2()` return just the id and the name of the snapshot and leaving the list of images that participated in the snapshot to this API.  All of this information is stored in the same `GroupSnapshot` structure in RADOS, so it gets fetched and decoded anyway.

Have you considered defining `rbd_group_snap_info2_t` as something like
```
typedef struct {
  char *id;
  rbd_snap_namespace_type_t unused;
  char *name;
  rbd_group_snap_state_t state;
  char *image_snap_name;
  size_t image_snaps_count;
  rbd_group_image_snap_info_t *image_snaps;
} rbd_group_snap_info2_t;
```
and returning that from both `rbd_group_snap_list2()` for all snapshots and `rbd_group_snap_get_info()` for a particular snapshot?

Reply:
Nice suggestion. I've implemented this.